kitops/modelkit/_pydantic_kit.py

```diff
@@ -0,0 +1,193 @@
+        name (Optional[str]): Identifier for the part.
+        type (Optional[str]): The type of the part (e.g. LoRA weights).
+    """
+
+    name: Optional[str] = ""
+    type: Optional[str] = ""
+
+
+class ModelSection(BasePathModel):
+    """
+    Details of the trained models included in the package.
+
+    Args:
+        path (str): Location of the model file or directory relative to the context.
+        name (Optional[str]): Name of the model.
+        framework (Optional[str]): AI/ML framework.
+        version (Optional[str]): Version of the model.
+        description (Optional[str]): Overview of the model.
+        license (Optional[str]): SPDX license identifier for the model.
+        parts (Optional[list[ModelPart]]): List of related files for the model (e.g. LoRA weights).
+        parameters (Optional[Any]): An arbitrary section of YAML that can be used to store any additional data that may
+            be relevant to the current model.
+    """
+
+    name: Optional[str] = ""
+    framework: Optional[str] = Field(default="", examples=["tensorflow", "pytorch", "onnx", "TensorRT"])
+    version: Optional[str] = Field(default="", examples=["0.0a13", "1.8.0"], coerce_numbers_to_str=True)
+    description: Optional[str] = ""
+    license: Optional[str] = ""
+    parts: Optional[list[ModelPart]] = []
+    parameters: Optional[Any] = Field(
+        default=None,
+        description=(
+            "An arbitrary section of YAML that can be used to store any additional data that may be relevant to the"
+            " current model, with a few caveats. Only a json-compatible subset of YAML is supported. Strings will be "
+            "serialized without flow parameters. Numbers will be converted to decimal representations (0xFF -> 255, "
+            "1.2e+3 -> 1200). Maps will be sorted alphabetically by key."
+        ),
+    )
+
+
+class PydanticKitfile(BaseModel):
+    """
+    Base class for the Pydantic Kitfile model.
+
+    Args:
+        manifestVersion (str): Specifies the manifest format version.
+        package (Optional[Package | dict]): This section provides general information about the AI/ML project.
+        code (Optional[list[CodeEntry]]): Information about the source code.
+        datasets (Optional[list[DatasetEntry]]): Information about the datasets used.
+        docs (Optional[list[DocsEntry]]): Information about included documentation for the model.
+        model (Optional[ModelSection | dict]): Details of the trained models included in the package.
+    """
+
+    manifestVersion: str = Field(default=..., examples=["1.0.0", "0.13a"], coerce_numbers_to_str=True)
+    code: Optional[list[CodeEntry]] = []
+    datasets: Optional[list[DatasetEntry]] = []
+    docs: Optional[list[DocsEntry]] = []
+    prop_package: Package = Field(default_factory=Package, alias="package", repr=False)
+    prop_model: ModelSection = Field(default_factory=lambda: ModelSection(path=""), alias="model", repr=False)
```

Comment:
These should be optional right?

Reply:
They _are_ optional but don't use the `Optional` type annotation since it's an alias for `object | None`. I took out the `Optional` type annotation as there were linting errors wrt to attribute assignments or attribute access. If `None` is an expected type for `Kitfile.package`, then attempt to set `Kitfile.package.name` would be invalid, which is what was being raised by the linter.